### DISCUSSION TITLE
Error: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: object.

### DISCUSSION THREAD

**Discussion by martyr-k** [remix-run/remix#3710] [Q&A]
I'm trying to redirect to a resource route from an action but I'm encountering an `Element type is invalid` error because the target route is being treated as a UI route. I know if you are navigating to a resource route via `<Link />` you add a `reloadDocument` prop to prevent this from happening. Is there anyway to replicate this using the `redirect` helper or HTTP response? 

This is what I have currently:

```js
return redirect(redirectTo);

// alternatively
return new Response(null, {
  status: 303,
  headers: {
    Location: redirectTo,
  },
});
```

Thanks!

**Comment by Mange** [comment]
I've raised a similar concern in #5221, but this used to work for form submissions by adding `reloadDocument` to the form:

```jsx
<Form action="/resource-that-will-redirect" reloadDocument>
</Form>
```
